### DISCUSSION TITLE
Transmisson of data stopped without notice

### DISCUSSION THREAD

**Discussion by tomuGo** [hbldh/bleak#457] [Q&A]
* bleak version: 0.10.0
* Python version:3.9
* Operating System:  win10
* BlueZ version (`bluetoothctl -v`) in case of Linux:

### Description
```
            async with BleakClient(address) as client:
                x = await client.is_connected()
                logger.info("Connected: {0}".format(x))
                if x:
                    self.label_status_value.setText("已连接")
                    self.connect_setting_log.appendPlainText("connect success")
                    await client.start_notify(ble_service_uuid, self.notification_handler)
                    await stop_event.wait()
                    print("Stopping notifications/indications...")
                    await client.stop_notify(ble_service_uuid)

                else:
                    self.label_status_value.setText("未连接")
                    self.connect_setting_log.appendPlainText("connect fail")
```
I'm sure the ble is still  transmitting,but the code will stoped after some time.and has no error on console,and print("Stopping notifications/indications...") not print

**Comment by tomuGo** [comment]
```   async with BleakClient(address) as client:
        x = await client.is_connected()
        logger.info("Connected: {0}".format(x))
        print(x)
        while True:
            await client.start_notify(CHARACTERISTIC_UUID, notification_handler)
```

when i used this code,data transmission after some time,appear problem.i try some way,but i don't know how to solved it .think u.
![image](https://user-images.githubusercontent.com/24789356/108629722-49fd7680-749c-11eb-9f51-66c649b8e5e6.png)

**Reply by hbldh** [reply]
You should not do a
```
while True:
    await client.start_notify(CHARACTERISTIC_UUID, notification_handler)
```
This will call the `start_notify` method infinitely, and this is possibly a reason for your `E_ABORT` situation. Calling it once is enough.

```
    async with BleakClient(address) as client:
        logger.info(f"Connected: {client.is_connected}")
        self.connect_setting_log.appendPlainText("connect success")
        await client.start_notify(ble_service_uuid, notification_handler)
        await stop_event.wait()
        print("Stopping notifications/indications...")
        await client.stop_notify(ble_service_uuid)
```

But unless you run a `stop_event.set()` somewhere, the code below `await stop_event.wait()` will never be run.

**Reply by hbldh** [reply]
Consider using a `disconnected_callback` as in [this example](https://github.com/hbldh/bleak/blob/develop/examples/disconnect_callback.py) so you can handle sudden disconnections.
